ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -1118,6 +1084,105 @@ module ClosestKdc = struct
       Xapi_periodic_scheduler.remove_from_queue periodic_update_task_name
 end
 
+module RotateMachinePassword = struct
+  let task_name = "Rotating machine password"
+
+  let tmp_krb5_conf =
+    Printf.sprintf "%s/pwrotate.krb5" (Filename.get_temp_dir_name ())
+
+  let kdc_fqdn_of_ip kdc =
+    let args =
+      ["ads"; "lookup"; "--server"; kdc; "--kerberos"; "-d"; debug_level ()]
+    in
+    Helpers.call_script !Xapi_globs.net_cmd args ~log_output:On_failure
+    |> Xapi_cmd_result.of_output ~sep:':' ~key:"Domain Controller"
+
+  let generate_krb5_tmp_config ~domain ~kdc_fqdn =
+    (* Configure which server to change the password
+     * https://web.mit.edu/kerberos/krb5-devel/doc/admin/conf_files/krb5_conf.html *)
+    let open Xapi_stdext_unix in
+    let realm = String.uppercase_ascii domain in
+    let conf_contents =
+      String.concat "\n"
+        [
+          "# auto-generated by xapi"
+        ; "[libdefaults]"
```

Comment:
If we generate a krb5.conf then we also need to specify the ciphersuites to accept here. Normally we'd rely on winbind to generate krb5.conf for us.

Reply:
Just updated with `include <winbind generated conf>`